src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployFeedJob.java

```diff
@@ -0,0 +1,133 @@
+                    if ((project.lastAutoDeploy == null ||
+                        feedVersion.dateCreated.after(project.lastAutoDeploy)) &&
+                        feedVersion.hasCriticalErrors()) {
+                        hasFeedVersionWithCriticalErrors = true;
+                        LOG.debug("Project {} contains feed version {} which has critical errors. Skipping auto deployment.",
+                            project.id,
+                            feedVersion.id);
+                        break;
+                    } else if ((project.lastAutoDeploy == null ||
+                        feedVersion.dateCreated.before(project.lastAutoDeploy)) &&
+                        deployment.hasFeedFetchesInProgress()) {
+                        hasFeedFetchesInProgress = true;
+                        LOG.debug("Project {} contains feed version {} which has active fetches in progress. Skipping auto deployment.",
+                            project.id,
+                            feedVersion.id);
+                        break;
+                    }
+                }
+
+                if (!hasFeedFetchesInProgress && !hasFeedVersionWithCriticalErrors) {
+                    // Send deployment (with new feed version) to most recently used server.
+                    if (deployment.latest() != null) {
+                        String latestServerId = deployment.latest().serverId;
+                        OtpServer server = Persistence.servers.getById(latestServerId);
+                        if (server != null) {
+                            // If there are no other fetches in progress, queue up the deploy job.
+                            if (DeploymentController.queueDeployJob(new DeployJob(deployment, owner, server))) {
+                                LOG.debug("Last auto deploy date updated for project id {}.", project.id);
+                                project.lastAutoDeploy = new Date();
+                                Persistence.projects.replace(project.id, project);
+                            } else {
+                                LOG.debug("Could not auto-deploy to {} due to conflicting active deployment for project id {}.",
+                                    server.name,
+                                    project.id);
+                            }
+                        } else {
+                            LOG.debug("Server with id {} no longer exists. Skipping deployment for project id {}.",
+                                latestServerId,
+                                project.id);
+                        }
+                    } else {
+                        // FIXME: Should we deploy some other server if deployment has not previously been deployed?
+                        LOG.debug("Deployment {} has never been deployed. Skipping auto-deploy for project id {}.",
+                            deployment.id,
+                            project.id);
+                    }
```

Comment:
I think this might end up being a gotcha as the other current deployments will have to have been ran at least once manually in order for there to be an association with the deployment to the server it was deployed to. Ending the job at this point is the right thing to do. However, I'd go ahead and bump the log level and send off a notification to be helpful to users.

Reply:
Updated.